### DISCUSSION TITLE
Help with config settings

### DISCUSSION THREAD

**Discussion by marcvl64** [troglobit/smcroute#195] [General]
I'm sorry if this is a newbie question, but I'm struggling to set up the right config for my use case: I have a Peplink transit cellular router installed in an RV. The local RV network runs a Home Assistant server and has a ton of sensors/devices, each with their own local IP address and name (192.168.50.xxx and zzzz.local). The router has a VPN connection to a virtual cloud-based router with static public IP. The cloud-based router is configured with a static route that forwards all incoming traffic to the local RV network. I'm trying to access that local network using an OpenVPN connection from my Mac to my cloud-based router. Doing so, I can get access to any local device by specifying its IP address (like 'ping 192.168.50.1' works great), but I cannot use hostnames ('ping homeassistant.local' doesn't resolve). I'm thinking installing and running smcroute on my Mac is the solution to this problem, but I'm not sure how to set up the config file so that multicasting works. I'm not a networking engineer and I'm not sure I understand completely how the group and route/source elements have to be used in the particular case. I would greatly appreciate if you could point me in the right direction.

**Comment by troglobit** [comment]
Hi, macOS does not have support for multicast routing in the kernel.  It can only act as an end device for multicast.

Setting up multicast routing in general is a non-trivial task.  It often requires deeper understanding of various network protocols, e.g., mDNS-SD in your case, their limitations and restrictions.  For instance, mDNS (.local addresses) is limited to LAN only and not supposed to be routed.

**Reply by marcvl64** [reply]
Ah ok. I was under the impression that your software would enable such routing. Sorry if I misunderstood. It’s strange to me that what looks like a trivial problem is not actually solvable.

**Reply by troglobit** [reply]
All the static and dynamic routing daemons for multicast only handle the userspace end of things, translating the user data input, or neighboring router commands, to the kernel's forwarding database.  macOS does not have a forwarding database for multicast, unfortunately.

There are possibly other ways to achieve what you want, but it's better to ask on stackoverflow or something.  Good luck!
